# Patch taken from the Gnome Flashback project to remove the default limit of 20 notifications at a time:
# https://gitlab.gnome.org/GNOME/gnome-flashback/-/commit/40c5db2e6b343c5afad78034a2fa93b9b000ef11
# See the following issue for more details:
# https://gitlab.gnome.org/GNOME/gnome-flashback/-/issues/75
diff --git a/src/nd-daemon.c b/src/nd-daemon.c
index 943e48a..5724405 100644
--- a/src/nd-daemon.c
+++ b/src/nd-daemon.c
@@ -36,8 +36,6 @@
 #define INFO_VERSION PACKAGE_VERSION
 #define INFO_SPEC_VERSION "1.2"
 
-#define MAX_NOTIFICATIONS 20
-
 struct _NdDaemon
 {
   GObject            parent;
@@ -180,24 +178,11 @@ handle_notify_cb (NdFdNotifications     *object,
                   gpointer               user_data)
 {
   NdDaemon *daemon;
-  const gchar *error_name;
-  const gchar *error_message;
   NdNotification *notification;
   gint new_id;
 
   daemon = ND_DAEMON (user_data);
 
-  if (nd_queue_length (daemon->queue) > MAX_NOTIFICATIONS)
-    {
-      error_name = "org.freedesktop.Notifications.MaxNotificationsExceeded";
-      error_message = _("Exceeded maximum number of notifications");
-
-      g_dbus_method_invocation_return_dbus_error (invocation, error_name,
-                                                  error_message);
-
-      return TRUE;
-    }
-
   if (replaces_id > 0)
     {
       notification = nd_queue_lookup (daemon->queue, replaces_id);
